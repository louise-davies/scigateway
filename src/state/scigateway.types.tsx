import { Step } from 'react-joyride';
import { Theme } from '@material-ui/core/styles/createMuiTheme';

export const AuthSuccessType = 'scigateway:auth_success';
export const AuthFailureType = 'scigateway:auth_failure';
export const LoadingAuthType = 'scigateway:loading_auth';
export const LoadedAuthType = 'scigateway:loaded_auth';
export const LoadAuthProviderType = 'scigateway:loading_auth_provider';
export const LoadScheduledMaintenanceStateType =
  'scigateway:load_scheduled_maintenance_state';
export const LoadMaintenanceStateType = 'scigateway:load_maintenance_state';
export const InvalidateTokenType = 'scigateway:api:invalidate_token';
export const ConfigureStringsType = 'scigateway:configure_strings';
export const LoginType = 'scigateway:login';
export const NotificationType = 'scigateway:api:notification';
export const RegisterRouteType = 'scigateway:api:register_route';
export const RequestPluginRerenderType = 'scigateway:api:plugin_rerender';
export const SendThemeOptionsType = 'scigateway:api:send_themeoptions';
export const LoadDarkModePreferenceType =
  'scigateway:load_dark_mode_preference';
export const SignOutType = 'scigateway:signout';
export const ToggleDrawerType = 'scigateway:toggledrawer';
export const DismissNotificationType = 'scigateway:dismissnotification';
export const ConfigureFeatureSwitchesType = 'scigateway:feature_switches';
export const AddNotificationsType = 'scigateway:add_notification';
export const SiteLoadingType = 'scigateway:site_loading';
export const ConfigureAnalyticsType = 'scigateway:configure_analytics';
export const InitialiseAnalyticsType = 'scigateway:initialise_analytics';
export const ToggleHelpType = 'scigateway:toggle_help';
export const AddHelpTourStepsType = 'scigateway:add_help_tour_steps';
export const RegisterStartUrlType = 'scigateway:register_start_url';

export interface NotificationPayload {
  message: string;
  severity: string;
}

export interface AddNotificationsPayload {
  notifications: NotificationPayload[];
}

export interface AppStrings {
  [id: string]: string;
}

export interface ConfigureStringsPayload {
  res: ApplicationStrings;
}

export interface ApplicationStrings {
  [section: string]: AppStrings;
}

export interface FeatureSwitchesPayload {
  switches: FeatureSwitches;
}

export interface FeatureSwitches {
  showContactButton: boolean;
  showHelpPageButton: boolean;
}

export interface StartUrlPayload {
  startUrl: string;
}

export interface RegisterRoutePayload {
  section: string;
  link: string;
  plugin: string;
  displayName: string;
  order: number;
  helpText?: string;
  helpSteps?: { target: string; content: string }[];
}

export interface PluginConfig {
  section: string;
  link: string;
  plugin: string;
  displayName: string;
  order: number;
  helpText?: string;
  helpSteps?: { target: string; content: string }[];
}

export interface GroupedPlugins {
  [section: string]: PluginConfig[];
}

export interface LoginPayload {
  username: string;
  password: string;
}

export interface AccessPayload {
  authorisation: boolean;
  accessToken: string;
  error: null;
}

export interface AuthorisedPayload {
  token: string;
}

export interface SendThemeOptionsPayload {
  theme: Theme;
  broadcast: boolean;
}

export interface LoadDarkModePreferencePayload {
  darkMode: boolean;
}

export interface DismissNotificationPayload {
  index: number;
}

export interface AuthProviderPayload {
  authProvider: string;
  authUrl: string | undefined;
}

export interface ScheduledMaintenanceStatePayLoad {
  scheduledMaintenance: ScheduledMaintenanceState;
}

export interface ScheduledMaintenanceState {
  show: boolean;
  message: string;
}

export interface MaintenanceStatePayLoad {
  maintenance: MaintenanceState;
}

export interface MaintenanceState {
  show: boolean;
  message: string;
}

export interface SiteLoadingPayload {
  loading: boolean;
}

export interface ConfigureAnalyticsPayload {
  id: string;
}

export interface AddHelpTourStepsPayload {
  steps: Step[];
}
