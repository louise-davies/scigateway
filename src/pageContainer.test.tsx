import React from 'react';
import { ReactWrapper, shallow } from 'enzyme';

import thunk from 'redux-thunk';
import configureStore from 'redux-mock-store';

// history package is part of react-router, which we depend on
// eslint-disable-next-line import/no-extraneous-dependencies
import { createLocation } from 'history';
import { MemoryRouter } from 'react-router-dom';

import PageContainer from './pageContainer.component';
import { StateType } from './state/state.types';
import { authState, initialState } from './state/reducers/scigateway.reducer';

describe('PageContainer - Tests', () => {
  let state: StateType;

  const createWrapper = (state: StateType): ReactWrapper => {
    const mockStore = configureStore([thunk]);
    return shallow(
      <MemoryRouter initialEntries={[{ key: 'testKey' }]}>
        <PageContainer store={mockStore(state)} />
      </MemoryRouter>
    );
  };

  beforeEach(() => {
    state = {
      scigateway: { ...initialState, authorisation: { ...authState } },
      router: { location: createLocation('/') },
    };
  });

  it('renders correctly', () => {
    const wrapper = createWrapper(state).find(PageContainer).dive();

    expect(wrapper).toMatchSnapshot();
  });
});
