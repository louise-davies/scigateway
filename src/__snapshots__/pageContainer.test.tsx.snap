// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageContainer - Tests renders correctly 1`] = `
<Styled(div)>
  <Connect(Preloader)
    fullScreen={true}
  />
  <Connect(MainAppBar) />
  <Connect(NavigationDrawer) />
  <Connect(Tour) />
  <Connect(CookieConsent) />
  <Connect(Routing) />
  <Connect(Footer) />
</Styled(div)>
`;
